code/planning/src/behavior_agent/behavior_tree.py

```
@@ -152,12 +151,21 @@ def update(self):
+
+    def _handle_parameter_group(self, group):
+        if self.config is None:
+            return
+        for param in group["parameters"]:
             param_name = param["name"]
             self.blackboard.set(
                 f"/params/{param_name}", self.config[param_name], overwrite=True
             )
-        return py_trees.common.Status.SUCCESS
+
+        for group in group["groups"]:
```

Ruff (0.8.2)

Loop control variable `group` overrides iterable it iterates

(B020)